src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -109,7 +109,9 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             _socketType = socketType;
             _protocolType = protocolType;
 
-            Debug.Assert(addressFamily != AddressFamily.InterNetworkV6 || !DualMode);
+            // TODO: I am not hitting this assertion in the desktop code,
```

Comment:
Thanks @mellinoe. I don't think the issue should block this PR.

Because you didn't get this on Desktop, I believe it may be a problem in how we changed this code in CoreFX.
Please open an issue for either me or @pgavlin to investigate and update the comment with the number.